src/renderer/components/editorWithTabs/tabs.vue

```diff
@@ -18,32 +27,131 @@
         </li>
-        <li class="new-file">
-          <svg class="icon" aria-hidden="true"
-            @click.stop="newFile()"
-          >
-            <use xlink:href="#icon-plus"></use>
-          </svg>
-        </li>
       </ul>
     </div>
+    <div
+      class="new-file"
+    >
+      <svg class="icon" aria-hidden="true"
+        @click.stop="newFile()"
+      >
+        <use xlink:href="#icon-plus"></use>
+      </svg>
+    </div>
+  </div>
 </template>
 
 <script>
   import { mapState } from 'vuex'
+  import autoScroll from 'dom-autoscroller'
+  import dragula from 'dragula'
   import { tabsMixins } from '../../mixins'
 
   export default {
+    data () {
+      return {
+        autoScroller: null,
+        drake: null
+      }
+    },
     mixins: [tabsMixins],
     computed: {
       ...mapState({
-        currentFile: state => state.editor.currentFile,
-        tabs: state => state.editor.tabs
+        'currentFile': state => state.editor.currentFile,
+        'tabs': state => state.editor.tabs,
+        'showSideBar': state => state.layout.showSideBar,
+        'sideBarWidth': state => state.layout.sideBarWidth
       })
     },
     methods: {
       newFile () {
         this.$store.dispatch('NEW_BLANK_FILE')
+      },
+      handleTabScroll (event) {
+        // Use mouse wheel value first but prioritize X value more (e.g. touchpad input).
+        let delta = event.deltaY
+        if (event.deltaX !== 0) {
+          delta = event.deltaX
+        }
+
+        const tabs = this.$refs.tabContainer
+        const newLeft = Math.max(0, Math.min(tabs.scrollLeft + delta, tabs.scrollWidth ))
+        tabs.scrollLeft = newLeft
+      }
+    },
+    mounted () {
+      this.$nextTick(() => {
+        const tabs = this.$refs.tabContainer
+
+        // Allow to scroll through the tabs by mouse wheel or touchpad.
+        tabs.addEventListener('wheel', this.handleTabScroll)
+
+        // Allow tab drag and drop to reorder tabs.
+        const drake = this.drake = dragula([ this.$refs.tabDropContainer ], {
+          direction: 'horizontal',
+          revertOnSpill: true,
+          mirrorContainer: this.$refs.tabDropContainer,
+          ignoreInputTextSelection: false
+        }).on('drop', (el, target, source, sibling) => {
+          // Current tab that was dropped and need to be reordered.
+          const droppedId = el.getAttribute('data-id')
+          // This should be the next tab (tab | ... | el | sibling | tab | ...) but may be
+          // the mirror image (tab | ... | el | sibling or null) if last tab.
+          // NOTE: According documentation sibling can be null --> last tab.
+          const nextTabId = sibling && sibling.getAttribute('data-id')
+          const isLastTab = !sibling || sibling.classList.contains('gu-mirror')
+          if (!droppedId || (sibling && !nextTabId)) {
+            throw new Error('Cannot reorder tabs: invalid tab id.')
+          }
+
+          const moveItem = (arr, from, to) => {
```

Comment:
Because we use the Vue framework, we can directly modify the order of the items in the array, and the corresponding changes will be made elsewhere, but the best practice still needs to dispatch an action through vuex, and then assign a new tabs array to the store. 

You can add an action in `store/editor.js`.